### PR TITLE
refactor(api/core): Improve type hints and apply ruff formatter in agent runner and model manager.

### PR DESCRIPTION
**Pull request by laipz8200**
# Checklist:

> [!IMPORTANT]  
> Please review the checklist below before submitting your pull request.

- [ ] Please open an issue before creating a PR or link to an existing issue
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

# Description

Improve type hints:

- change some `list` to `Sequence`

Format:

- Apply ruff formatter.

Fixes 

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [x] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [ ] Dependency upgrade

# Testing Instructions

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

### CODE REVIEW

**Review by bowenliang123 — APPROVED** [review]
LGTM

File: api/core/agent/base_agent_runner.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import json
 import logging
 import uuid
+from collections.abc import Mapping, Sequence
```
Comment (bowenliang123) [inline api/core/agent/base_agent_runner.py:4]: BTW, `Mapping` and `Sequence` are abstract types but not typing , which is related to implementation.

Comment (laipz8200) [reply api/core/agent/base_agent_runner.py:4]: Thanks for the reminder! Using them here is to remind everyone that this container is immutable.

Comment (bowenliang123) [reply api/core/agent/base_agent_runner.py:4]: Sure. It's not a blocker issue.

**Review by crazywoola — APPROVED** [review]

**Pull request merged**